release/util.bzl

```diff
@@ -1,79 +1,51 @@
             "//templates:templates-tarball.tar.gz",
-            "//triggers/daml:daml-trigger-dars",
-            "//daml-script/daml:daml-script-dars",
-            "//daml-assistant/daml-sdk:sdk_deploy.jar",
         ],
-        outs = ["{}.tar.gz".format(name)],
-        cmd = """
-          # damlc
-          VERSION={version}
-          OUT=sdk-$$VERSION
-          mkdir -p $$OUT
-
-          cp $(location //:NOTICES) $$OUT/NOTICES
-
-          cp $(location :install.sh) $$OUT/install.sh
-          cp $(location :install.bat) $$OUT/install.bat
-
-          cp $(location :sdk-config.yaml.tmpl) $$OUT/sdk-config.yaml
-          sed -i "s/__VERSION__/$$VERSION/" $$OUT/sdk-config.yaml
-
-          mkdir -p $$OUT/daml
-          tar xf $(location //daml-assistant:daml-dist) --strip-components=1 -C $$OUT/daml
-
-          mkdir -p $$OUT/damlc
-          tar xf $(location //compiler/damlc:damlc-dist) --strip-components=1 -C $$OUT/damlc
-
-          mkdir -p $$OUT/daml-libs
-          cp -t $$OUT/daml-libs $(locations //triggers/daml:daml-trigger-dars)
-          cp -t $$OUT/daml-libs $(locations //daml-script/daml:daml-script-dars)
-
-          mkdir -p $$OUT/daml-helper
-          tar xf $(location //daml-assistant/daml-helper:daml-helper-dist) --strip-components=1 -C $$OUT/daml-helper
-
-          mkdir -p $$OUT/daml2ts
-          tar xf $(location //language-support/ts/codegen:daml2ts-dist) --strip-components=1 -C $$OUT/daml2ts
-
-          mkdir -p $$OUT/studio
-          cp $(location //compiler/daml-extension:vsix) $$OUT/studio/daml-bundled.vsix
-
-          mkdir -p $$OUT/templates
-          tar xf $(location //templates:templates-tarball.tar.gz) --strip-components=1 -C $$OUT/templates
-
-          mkdir -p $$OUT/daml-sdk
-          cp $(location //daml-assistant/daml-sdk:sdk_deploy.jar) $$OUT/daml-sdk/daml-sdk.jar
-          cp -L $(location //ledger-service/http-json:release/json-api-logback.xml) $$OUT/daml-sdk/
-          cp -L $(location //ledger/sandbox:src/main/resources/logback.xml) $$OUT/daml-sdk/sandbox-logback.xml
-          cp -L $(location //navigator/backend:src/main/resources/logback.xml) $$OUT/daml-sdk/navigator-logback.xml
-          cp -L $(location //extractor:src/main/resources/logback.xml) $$OUT/daml-sdk/extractor-logback.xml
-          cp -L $(location //language-support/java/codegen:src/main/resources/logback.xml) $$OUT/daml-sdk/codegen-logback.xml
-          cp -L $(location //triggers/runner:src/main/resources/logback.xml) $$OUT/daml-sdk/trigger-logback.xml
-          cp -L $(location //daml-script/runner:src/main/resources/logback.xml) $$OUT/daml-sdk/script-logback.xml
-
-          tar c --format=ustar --sort=name --owner=root:0 --group=root:0 --mtime=0 $$OUT | gzip -n > $@
-        """.format(version = version),
-        visibility = ["//visibility:public"],
+        strip_prefix = "/",
+        package_dir = "sdk-{version}".format(version = version),
+        remap_paths = {
```

Comment:
Do we rely on the top-level directory for things like the damlc/assistant/… tarballs now? Maybe worth leaving a comment here then since the top-level directory and the path in the release tarball did not always match up in the past.

Reply:
Turns out that the tarballs generated by `pkg_tar` are subtly different from the previous ones. `pkg_tar` prefixes all entries with a `./`. This doesn't change the result of unpacking the archive, but it changes the meaning of `tar`'s `--strip-components` flag, which breaks the `daml-sdk-head` script, and may impact users. The underlying tool that is called by `pkg_tar` has a flag to change the overall path prefix. I tried patching the rule to make use of that flag to prevent the `./` prefix. However, unfortunately that prefix is used by `pkg_tar` when combining multiple tarballs, without it files in `deps` do not get prepended with `package_dir` (i.e. `sdk-X.Y.Z`). So, `pkg_tar` seems unfit for our use-case. I've reverted to `genrule` with `tar`, but added the flags for reproducibility.